app/src/androidTest/java/ch/epfl/sdp/storage/ProbabilityStorageTest.java

```diff
@@ -0,0 +1,222 @@
+package ch.epfl.sdp.storage;
+
+import androidx.test.rule.ActivityTestRule;
+
+import org.junit.BeforeClass;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+
+import java.io.IOException;
+import java.text.DateFormat;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Collections;
+import java.util.Date;
+import java.util.Map;
+
+import ch.epfl.sdp.CoronaGame;
+import ch.epfl.sdp.MainActivity;
+import ch.epfl.sdp.TestTools;
+
+import static org.hamcrest.CoreMatchers.equalTo;
+import static org.hamcrest.MatcherAssert.assertThat;
+
+public class ProbabilityStorageTest {
+
+    @Rule
+    public final ActivityTestRule<MainActivity> mActivityRule = new ActivityTestRule<>(MainActivity.class);
+
+    @Rule
+    public ExpectedException testException = ExpectedException.none();
+
+    private static final String TEST_FILENAME = "history_test_file.csv";
+
+    private StorageManager<Integer, Double> getIntDoubleManager() {
+        return new ConcreteManager<>(
+                CoronaGame.getContext(),
+                TEST_FILENAME,
+                Integer::valueOf,
+                Double::valueOf);
+    }
+
+    private DateFormat format = new SimpleDateFormat("E MMM dd HH:mm:ss zzz yyyy");
+
+    private StorageManager<Date, Double> getDateDoubleManager() {
+        return new ConcreteManager<>(
+                CoronaGame.getContext(),
+                TEST_FILENAME,
+                k -> {
+                    try {
+                        return format.parse(k);
+                    } catch (ParseException e) {
+                        throw new IllegalArgumentException();
+                    }
+                },
+                Double::valueOf);
+    }
+
+    @BeforeClass
+    public static void DeleteTestFile() {
+        new ConcreteManager<>(
+                CoronaGame.getContext(),
+                TEST_FILENAME,
+                k -> 0,
+                v -> null
+        ).delete();
+    }
+
+    @Test
+    public void fileCanBeReadAndWritten() {
+        StorageManager<Integer, Double> manager = getIntDoubleManager();
```

Comment:
Does it really test the file read and write? It looks that the creation and deletion are tested to not produce exception but not that the file are read and wrote correctly to the StorageManager

Reply:
Agree, the name of the test is no more appropriate